packages/@aws-cdk/aws-synthetics-alpha/lib/code.ts

```diff
@@ -164,12 +164,7 @@ export class InlineCode extends Code {
     }
   }
 
-  public bind(_scope: Construct, handler: string, _family: RuntimeFamily): CodeConfig {
-
-    if (handler !== 'index.handler') {
-      throw new Error(`The handler for inline code must be "index.handler" (got "${handler}")`);
-    }
```

Comment:
I'm concerned about removing this. Have you tested this to make sure that it works? IIRC lambda will bundle inline code into an `index.js|py` file, which means that it should at least always start with `index.`